apps/web/components/getting-started/steps-views/UserSettings.tsx

```diff
@@ -1,120 +1,198 @@
-  hideUsername?: boolean;
 }
 
 const UserSettings = (props: IUserSettingsProps) => {
+  const [selectedOption, setSelectedOption] = useState<AccountOption>("personal");
   const { nextStep } = props;
   const [user] = trpc.viewer.me.useSuspenseQuery();
   const { t } = useLocale();
-  const { setTimezone: setSelectedTimeZone, timezone: selectedTimeZone } = useTimePreferences();
+  const router = useRouter();
   const telemetry = useTelemetry();
-  const userSettingsSchema = z.object({
-    name: z
-      .string()
-      .min(1)
-      .max(FULL_NAME_LENGTH_MAX_LIMIT, {
-        message: t("max_limit_allowed_hint", { limit: FULL_NAME_LENGTH_MAX_LIMIT }),
-      }),
-  });
-  const {
-    register,
-    handleSubmit,
-    formState: { errors },
-  } = useForm<z.infer<typeof userSettingsSchema>>({
-    defaultValues: {
-      name: user?.name || "",
+  const createEventType = trpc.viewer.eventTypes.create.useMutation();
+  const createSchedule = trpc.viewer.availability.schedule.create.useMutation();
+  const updateProfile = trpc.viewer.updateProfile.useMutation();
+  const isPending = createEventType.isPending || createSchedule.isPending || updateProfile.isPending;
+  const options = [
+    {
+      value: "personal",
+      icon: "user",
+      title: t("for_personal_use"),
+      description: t("for_personal_use_description"),
+      disabled: false,
+      tooltip: null,
+    },
+    {
+      value: "team",
+      icon: "users",
+      title: t("with_my_team"),
+      description: t("with_my_team_description"),
+      disabled: false,
+      tooltip: null,
+    },
+    {
+      value: "org",
+      icon: "building",
+      title: t("for_my_organization"),
+      description: t("for_my_organization_description"),
+      disabled: true,
+      tooltip: t("contact_sales"),
     },
-    reValidateMode: "onChange",
-    resolver: zodResolver(userSettingsSchema),
-  });
+  ] as const;
+
+  const DEFAULT_EVENT_TYPES = [
+    {
+      title: t("15min_meeting"),
+      slug: "15min",
+      length: 15,
+    },
+    {
+      title: t("30min_meeting"),
+      slug: "30min",
+      length: 30,
+    },
+    {
+      title: t("secret_meeting"),
+      slug: "secret",
+      length: 15,
+      hidden: true,
+    },
+  ] as const;
 
   useEffect(() => {
     telemetry.event(telemetryEventTypes.onboardingStarted);
   }, [telemetry]);
 
-  const utils = trpc.useUtils();
-  const onSuccess = async () => {
-    await utils.viewer.me.invalidate();
-    nextStep();
+  function onError(error: TRPCClientErrorLike<any>) {
+    console.error(error);
+    showToast(t("something_went_wrong"), "error");
+  }
+
+  const createDefaultAvailabilityAndEventTypes = async ({ onSuccess }: { onSuccess: () => void }) => {
+    try {
+      await Promise.all([
+        ...DEFAULT_EVENT_TYPES.map((event) => createEventType.mutateAsync(event)),
+        createSchedule.mutateAsync({
+          name: t("default_schedule_name"),
+          ...DEFAULT_SCHEDULE,
+        }),
```

Comment:
@emrysal I see that you removed the length check here. If user somehow goes back to `/getting-started` after completing onboarding, default event types and availabilities will be created again

Reply:
I suggest we create methods to repositories so that we have:
- EventTypesRepository.createDefault()
- AvailabilityRepository.createDefault()

and in the methods themselves, we can check if they already exist